bigchaindb_driver/transport.py

```diff
@@ -50,11 +74,28 @@ def forward_request(self, method, path=None,
 
         """
         connection = self.get_connection()
-        response = connection.request(
```

Comment:
We need to accept a timeout here - driver users should be able to specify for how long they are able to wait for the response. Consequently, a timeout has to be specified for each HTTP request.

Here are some code snippets which I hope are helpful:
```
time_left = datetime.timedelta(timeout)
start = time.time()
while True:
    connection = self.get_connection()

    if connection is not None:
        response = self._request(..timeout=time_left.seconds)
        if response is not None:
            return response
    
    end = time.time()
    time_left -= datetime.timedelta(seconds=int(end-start))
    if time_left.total_seconds() <= 0:
        raise ConnectionTimeoutError  # or smth similar
```
`def _request`:

```
try:
    response = connection.request(..timeout=timeout)
except Exception as e:
    log.warning(f'An exception occurred while connecting to the node {path}: {e}')
    self.pool.fail_node()
    return
else:
    self.pool.success_node()
    return response.data
```
Note the logging - you can instantiate a logger in the global scope via `log = logging.getLogger(__name__)`.

Also, I simplified error handing in these snippets - please, tell me if I missed anything or if there are any other concerns to take into account. For instance, what is the purpose of passing `error` to `forward_request`?

Reply:
thanks for the snippets, implemented on the next push with a equivalent mindset, and updated from recursive to iterative approach. 
also about:
> Also, I simplified error handing in these snippets - please, tell me if I missed anything or if there are any other concerns to take into account. For instance, what is the purpose of passing error to forward_request?

is the last error we obtained from the set of requests executed during the loop, some tests will not pass if we raise timeout exception for all posible exceptions raised, so the last exception raised will have more priority over the timeout, and the goal is to inform (give a hint) the user what is wrong with the last node used.